src/library_html5.js

```diff
@@ -3269,4 +3237,56 @@ var LibraryJSEvents = {
   }
 };
 
+function handleWebGLProxying(funcs) {
```

Comment:
Please add some docs for this, maybe here.

In particular it would be good to note that `sync_on_webgl_context_handle_thread` is a special proxying mode only used in this file, and removed while the file is processed so nothing outside will ever see it.